### ISSUE TITLE
fail to convert EDS from MRS with broken HCONS

### ISSUE THREAD

**Issue by arademaker** [delph-in/pydelphin#319]
```
[ TOP: h0
  INDEX: e2 [ e SF: prop TENSE: tensed MOOD: indicative PROG: - PERF: - ]
  RELS: < [ _communicate_v_to<0:11> LBL: h1 ARG0: e4 [ e SF: prop TENSE: tensed MOOD: indicative PROG: - PERF: - ] ARG1: i3 ARG2: h5 ARG3: i6 ]
          [ _or_c<12:14> LBL: h1 ARG0: e2 ARG1: e4 ARG2: e7 [ e SF: prop TENSE: pres MOOD: indicative PROG: - PERF: - ] ]
          [ _express_v_to<15:22> LBL: h1 ARG0: e7 ARG1: i3 ARG2: h8 ARG3: i9 ]
          [ unknown<23:33> LBL: h10 ARG: u12 ARG0: e11 [ e SF: prop TENSE: untensed MOOD: indicative ] ]
          [ _by_p_means<23:25> LBL: h10 ARG0: e11 ARG1: u13 ARG2: x14 ]
          [ udef_q<26:33> LBL: h15 ARG0: x14 RSTR: h16 BODY: h17 ]
          [ nominalization<26:33> LBL: h18 ARG0: x14 ARG1: h19 ]
          [ _write_v_to<26:33> LBL: h19 ARG0: e20 [ e SF: prop TENSE: untensed MOOD: indicative PROG: + PERF: - ] ARG1: i21 ARG2: i22 ] >
  HCONS: < h0 qeq h1 h5 qeq h23 h8 qeq h23 h16 qeq h18 > ]
```

```
....
Traceback (most recent call last):
  File "process.py", line 20, in <module>
    e = eds.from_mrs(m)
  File "/Users/ar/venv-dev/lib/python3.8/site-packages/delphin/eds/_operations.py", line 45, in from_mrs
    deps = _mrs_args_to_basic_deps(m, hcmap, reps)
  File "/Users/ar/venv-dev/lib/python3.8/site-packages/delphin/eds/_operations.py", line 95, in _mrs_args_to_basic_deps
    tgt = reps[lbl][0].id
KeyError: 'h23'
```

**Comment by arademaker** [comment]
I have also reported in https://github.com/delph-in/erg/issues/25. It looks like bug in the ERG.

**Comment by goodmami** [comment]
I see the MRS and the traceback, but no code. Is this just using the `delphin convert` command?

Also note that conversion *from* EDS to MRS is not supported as the conversion from MRS *to* EDS is lossy.

**Comment by arademaker** [comment]
The error was using EDM

```
(venv-dev) ar@leme wn % delphin edm -vvv ../terg/tsdb/gold/omw omw-parsed
```

**Comment by goodmami** [comment]
What is, or should be, the expected behavior when a qeq selects a label is not the label of any EP, or when half the EPS are disconnected from the top?

**Comment by arademaker** [comment]
I don’t know... maybe at least a proper error handling? A test of consistency before attempting the conversion? Btw, the doc string says the EDSError may be fired, but it is not true. The function does not have any raise command.

**Comment by goodmami** [comment]
I think I'd prefer to keep the behavior that an error is raised on ill-formed MRSs but, you're right, it should be raising more descriptive EDSErrors from the underlying errors. Any kind of robustness to ill-formed inputs should be optional. Also my general strategy here is to *not* do the costly consistency checks before conversion, and instead just fail during conversion (ideally with descriptive errors).

**Comment by arademaker** [comment]
This is related to #316. The origin of both errors is a broken MRS with `h?` that occurs only as arguments of EP, mentioned in the HCONS but not labeling any EP.  I pushed one more commit in the https://github.com/delph-in/pydelphin/pull/318 solving this error. I followed your approach in the `_mrs_args_to_basic_deps` function to just `continue` in the case of a missing key in the representatives map.
